File: email-backup.java/src/main/java/demo/OptionsBuilder.java
package demo;

import org.apache.commons.cli.OptionBuilder;
import org.apache.commons.cli.Options;

final public class OptionsBuilder {

    @SuppressWarnings("static-access")
    public static Options build() {
        Options options = new Options();

        options.addOption(OptionBuilder
                .withType(String.class)
                .withArgName("email")
                .hasArg()
                .isRequired()
                .withDescription("email sender/recipient")
                .create("email"));

        options.addOption(OptionBuilder
                .withType(String.class)
                .withArgName("password")
                .hasArg()
                .isRequired()
                .withDescription("email password")
                .create("password"));

        options.addOption(OptionBuilder
                .withType(String.class)
                .withArgName("host")
                .hasArg()
                .isRequired()
                .withDescription("SMTP host")
                .create("host"));

        options.addOption(OptionBuilder
                .withType(Number.class)
                .withArgName("port")
                .hasArg()
                .isRequired()
                .withDescription("SMTP port")
                .create("port"));

        options.addOption(OptionBuilder
                .withType(String.class)
                .withArgName("extension1,extension2,...")
                .hasArgs()
                .withValueSeparator(',')
                .isRequired(false)
                .withDescription("excluded file extensions (optional)")
                .create("excluded_extensions"));

        options.addOption(OptionBuilder
                .withType(String.class)
                .withArgName("file1,file2,...")
                .hasArgs()
                .withValueSeparator(',')
                .isRequired(false)
                .withDescription("included files (optional)")
                .create("included_files"));

        options.addOption(OptionBuilder
                .withType(String.class)
                .withArgName("email1,email2,...")
                .hasArgs()
                .withValueSeparator(',')
                .isRequired(false)
                .withDescription("additional recipients (optional)")
                .create("emails_to"));

        return options;
    }
}

File: email-backup.kotlin/src/main/kotlin/demo/OptionsBuilder.kt
package demo

import org.apache.commons.cli.OptionBuilder
import org.apache.commons.cli.Options

public object OptionsBuilder {

    @SuppressWarnings("static-access")
    public fun build(): Options {
        val options = Options()

        OptionBuilder.withType(String::class.java)
        OptionBuilder.withArgName("email")
        OptionBuilder.hasArg()
        OptionBuilder.isRequired()
        OptionBuilder.withDescription("email sender/recipient")
        options.addOption(OptionBuilder.create("email"))

        OptionBuilder.withType(String::class.java)
        OptionBuilder.withArgName("password")
        OptionBuilder.hasArg()
        OptionBuilder.isRequired()
        OptionBuilder.withDescription("email password")
        options.addOption(OptionBuilder.create("password"))

        OptionBuilder.withType(String::class.java)
        OptionBuilder.withArgName("host")
        OptionBuilder.hasArg()
        OptionBuilder.isRequired()
        OptionBuilder.withDescription("SMTP host")
        options.addOption(OptionBuilder.create("host"))

        OptionBuilder.withType(Number::class.java)
        OptionBuilder.withArgName("port")
        OptionBuilder.hasArg()
        OptionBuilder.isRequired()
        OptionBuilder.withDescription("SMTP port")
        options.addOption(OptionBuilder.create("port"))

        OptionBuilder.withType(String::class.java)
        OptionBuilder.withArgName("extension1,extension2,...")
        OptionBuilder.hasArgs()
        OptionBuilder.withValueSeparator(',')
        OptionBuilder.isRequired(false)
        OptionBuilder.withDescription("excluded file extensions (optional)")
        options.addOption(OptionBuilder.create("excluded_extensions"))

        OptionBuilder.withType(String::class.java)
        OptionBuilder.withArgName("file1,file2,...")
        OptionBuilder.hasArgs()
        OptionBuilder.withValueSeparator(',')
        OptionBuilder.isRequired(false)
        OptionBuilder.withDescription("included files (optional)")
        options.addOption(OptionBuilder.create("included_files"))

        OptionBuilder.withType(String::class.java)
        OptionBuilder.withArgName("email1,email2,...")
        OptionBuilder.hasArgs()
        OptionBuilder.withValueSeparator(',')
        OptionBuilder.isRequired(false)
        OptionBuilder.withDescription("additional recipients (optional)")
        options.addOption(OptionBuilder.create("emails_to"))

        return options
    }
}
